frame/composable-traits/src/staking_rewards.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{
+	financial_nft::{NFTClass, NFTVersion},
+	math::SafeSub,
+	time::{DurationSeconds, Timestamp},
+};
+use codec::{Decode, Encode};
+use core::fmt::Debug;
+use frame_support::{dispatch::DispatchResult, traits::Get};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::AtLeast32BitUnsigned, DispatchError, Perbill, SaturatedConversion};
+
+pub enum PositionState {
+	/// The position is not being rewarded yet and waiting for the next epoch.
+	Pending,
+	/// The position is currently locked and being rewarded.
+	LockedRewarding,
+	/// The position expired but still being rewarded.
+	Expired,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub enum ClaimStrategy {
+	/// Basic, canonical claiming.
+	Canonical,
+	/// Force claiming the reward by restaking if the position expired.
+	RestakeOnExpiry,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct StakingConfig<AccountId, DurationPresets, Rewards> {
+	/// The possible locking duration.
+	pub duration_presets: DurationPresets,
+	/// The assets we can reward stakers with.
+	pub rewards: Rewards,
+	/// The penalty applied if a staker unstake before the end date.
+	pub early_unstake_penalty: Perbill,
+	/// The beneficiary of the penalty.
+	pub penalty_beneficiary: AccountId,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct StakingTag<AccountId, CollectedRewards> {
+	/// The account that actually tagged the nft.
+	pub tagger: AccountId,
+	/// The beneficiary of the rewards distributed after tagging the position.
+	pub beneficiary: AccountId,
+	/// The rewards collected so far, rewards collected after this point will be distributed to the
+	/// `beneficiary`.
+	pub collected_rewards: CollectedRewards,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct StakingNFT<AccountId, AssetId, Balance, CollectedRewards> {
```

Comment:
The naming here stutters a bit, we have a type named `staking_rewards::StakingTag`, which I think would nicer as `staking_rewards::Tag`, locally we can then rename at import if it conflicts.

Reply:
Tagging feature has been removed